packages/esm-form-entry-app/src/app/form-data-source/form-data-source.service.ts

```diff
@@ -114,6 +117,56 @@ export class FormDataSourceService {
     );
   }
 
+  public getMostRecentObsValueBefore(date: string, patientUuid: string) {
+    type ObsResult = FHIRResource['resource'] & {
+      conceptUuid: string;
+      dataType?: string;
+    };
+
+    function isUuid(input: string) {
+      return input.length === 36;
+    }
+
+    const dataSource = {
+      obsValues: [],
+    };
+
+    const concepts = this.conceptResourceService.getAllConcepts();
+    concepts.subscribe((concepts) => {
+      const uuids = concepts.map((c) => c.uuid).join(',');
+
+      return this.observationResourceService.getMostRecentObsValues(date, uuids, patientUuid).subscribe((data) => {
+        const observations =
+          data?.entry?.map((entry) => {
+            const observation: ObsResult = {
+              ...entry.resource,
+              conceptUuid: entry.resource.code.coding.filter((c) => isUuid(c.code))[0]?.code,
```

Comment:
It would probably be more robust to just check for the first code element that doesn't have a `system` property, if this needs to be here... I'm a little confused as to why we need this at all, though...

Reply:
The returned obs data had many things and was an object so I wanted to return an array with formatted values, @ibacher please advise whether we need to format the data anyway, if we do, what is the expected JSON structure?